### PR TITLE
DOC: change colormap away from "jet" in examples

### PR DESCRIPTION
**Pull request by endolith**
1. `jet` is generally discouraged as perceptually misleading: https://github.com/matplotlib/matplotlib/issues/875#issue-4616667
2. I think it's more clear when all 3 images use the same colormap, to show how an edge detector converts a solid block of brightness into bright lines on just the edges:

```
          ______
    _____/      \_____  -->  _____/\____/\_____
```

### CODE REVIEW

**Comment by blink1073** [comment]
I like `YlGnBu_r` in the short term (for all previous uses of `jet`), followed by whatever the matplotlib folks settle on for their default.  Too bad `YlGnBu_r` doesn't have a snazzier name.  @tacaswell, you may want to consider giving your default map a name that is easy to remember (and say), like `jet` was.

**Comment by endolith** [comment]
I think `viewer_examples/plugins/watershed_demo.py` only works with `jet`?  It's using it for labeled colors, not as a heatmap?

**Comment by jni** [comment]
@endolith I just noticed that! Jet is *also* inappropriate to show arbitrary label maps. Is there anything it *can* do? =P

For label maps, I think the best approach is a random colormap in Lab space, like I once did [here](https://github.com/janelia-flyem/gala/blob/master/gala/viz.py#L50)

**Comment by endolith** [comment]
Well I think that's what the qualitative colormaps are meant for?  But they're currently implemented as LinearSegmentedColormaps and should have been ListedColormap?  https://github.com/matplotlib/matplotlib/issues/881

**Comment by endolith** [comment]
Also I'm confused because it seems like `skimage.feature` and `skimage.filters` will be combined into `skimage.filter`, but the documentation follows this and the actual code doesn't yet?  http://stackoverflow.com/a/26877466/125507  Like watershed_demo has `from skimage import filters` and then `edge_image = filter.sobel(viewer.image)`?

**Comment by jni** [comment]
@endolith no, the two modules aren't merging. Canny was moved from `filter` to `feature`, and `filter` will be renamed `filters`. `filter.sobel` is a bug, missed in the rename. Can you point me to the exact line in the codebase?

Regarding colormaps, I don't have all that much experience in this space... Is there a qualitative map that can be used for an arbitrary number of segments?

**Comment by blink1073** [comment]
@endolith, I think we need @tonysyu to weigh in on whether to change to a `ListedColormap`, since he implemented the [LinearColormap](https://github.com/scikit-image/scikit-image/blob/master/skimage/viewer/utils/core.py#L88).

**Comment by endolith** [comment]
@jni  Oh, ok.  The one I saw is https://github.com/scikit-image/scikit-image/blob/master/viewer_examples/plugins/watershed_demo.py#L31

grep finds some others to check:

https://github.com/scikit-image/scikit-image/blob/master/skimage/morphology/grey.py#L44
https://github.com/scikit-image/scikit-image/blob/master/skimage/morphology/grey.py#L105
https://github.com/scikit-image/scikit-image/blob/master/skimage/filters/tests/test_filter_import.py#L7

**Comment by tonysyu** [comment]
I think it's a good idea to define some `ListedColormap`s for this purpose. The `LinearColormap` is a nice easy way to get color palettes from certain colormaps (e.g. the `cm.Set*` colormaps), but now that you mention it, a better alternative would be to sample the colormap and save it as a real qualitative colormap; e.g.:
```python
qcmap_jet = ListedColormap(plt.cm.jet(np.linspace(0, 1, 5)))
```

**Comment by stefanv** [comment]
I guess we need to wait for matplotlib to finalize their decision, but at least we have alternatives now at http://bids.github.io/colormap/

**Comment by blink1073** [comment]
I think should include the viridia look up table in `utils` and perhaps set it upon import of skimage if jet is the current setting.

**Comment by stefanv** [comment]
I would not override the matplotlib defaults, in case the user chose to use something different.  But we can make sure that, when io.imshow is called and the matplotlib backend is in place, viridis is used.  Perhaps viridis can be exposed even more prominently so that users can use it easily?

**Comment by blink1073** [comment]
skimage.set_cmap?



—

**Comment by stefanv** [comment]
I shy away from carrying state.  It often leaves an awful mess in notebooks, e.g.

**Comment by blink1073** [comment]
How about if we register the [cmap](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/cm.py#L95), on import, and then use it by string literal name by default in `imshow` and all of our examples?

**Comment by stefanv** [comment]
That works!

**Comment by endolith** [comment]
1. This PR is just for changing the examples, not the whole package's default.
2. Yes, some of these should be using ListedColormap instead of LinearSegmentedColormap, but https://github.com/matplotlib/matplotlib/pull/3973 isn't changing for a while, so there isn't a convenient way to do this in examples yet. Is this patch acceptable as is for now?

**Comment by blink1073** [comment]
@endolith, this is also about a consistent message throughout the library.  We'd like to present a clear alternative to `jet` that has superior visual perception properties.

**Comment by blink1073** [comment]
@endolith, it looks like `nipy_spectral` was added in a later version of matplotlib, which is why the Python 2.7 build is failing.

**Comment by blink1073** [comment]
Perhaps just use `cubehelix` instead of `nipy_spectral` and we'll call it a day on this one?

**Comment by endolith** [comment]
Oh.  `nipy_spectral` is a synonym for `spectral` that I made to distinguish it from `Spectral`.  In the future it should be used, but if it's making the tests fail I can just change it back.

**Comment by endolith** [comment]
Ok, rebased and removed nipy_spectral for now

**Comment by stefanv** [comment]
Some of these should probably use the new colormap proposed in #1599

**Comment by soupault** [comment]
/ping @endolith 
Hi! Could you, please, rebase once again? We should really finalize this.

**Comment by endolith** [comment]
ok, rebased

File: doc/examples/segmentation/plot_watershed.py [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
 ax0.imshow(image, cmap=plt.cm.gray, interpolation='nearest')
 ax0.set_title('Overlapping objects')
-ax1.imshow(-distance, cmap=plt.cm.jet, interpolation='nearest')
+ax1.imshow(-distance, cmap=plt.cm.gray, interpolation='nearest')
```
Comment (soupault) [inline doc/examples/segmentation/plot_watershed.py]: I'd prefer to use non-grayscale colormap here, as it better presents what is actually the result.

**Comment by soupault** [comment]
Thanks @endolith ! One minor comment from my side and we're ready to go.

File: doc/examples/segmentation/plot_watershed.py (hunk 1 above)
Comment (endolith) [inline doc/examples/segmentation/plot_watershed.py]: What do you mean?

figure_1

Comment (soupault) [inline doc/examples/segmentation/plot_watershed.py]: Nevermind, gray is probably one of the best options here.
I wanted to get terrain-like effect, but to get one the image with higher resolution and histogram equalization are required.

**Comment by soupault** [comment]
:+1:

**Comment by jni** [comment]
Merging this, thank you @endolith!

@scikit-image/core if we want to make some examples use viridis, we can use a different PR for that.

**Pull request merged**